### PR TITLE
bump github.com/Azure/azure-sdk-to-go/sdk/resourcemanager/containerservice/armcontainerservice/v4 to v4.4.0-beta.2

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Updates AKS to the latest beta API. See release notes here: https://github.com/Azure/azure-sdk-for-go/releases?q=armcontainerservice&expanded=true

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.1
 	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2 v2.1.1
 	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v5 v5.2.0
-	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v4 v4.3.0
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v4 v4.4.0-beta.1
```
Comment (CecileRobertMichon) [inline go.mod:12]: /hold

I don't know that we want to bump to beta releases of the SDK. What are the support guarantees around beta releases? Are they subject to breaking changes?

related to #2625

Comment (mboersma) [reply go.mod:12]: I'm not sure we want to do this either; I created this PR partially to raise the question and partially to see if there are any technical issues for CAPZ using the beta release.

**Comment by mboersma** [comment]
/retitle bump github.com/Azure/azure-sdk-to-go/sdk/resourcemanager/containerservice/armcontainerservice/v4 to v4.4.0-beta.2

**Comment by mboersma** [comment]
/close

See #4069

**Pull request closed without merging**